### DISCUSSION TITLE
When does ArgoCD update/cleanup Prometheus Metrics?

### DISCUSSION THREAD

**Discussion by reefland** [argoproj/argo-cd#9929] [Q&A]
ArgoCD `v2.4.4+fe80bdc` Installed via Helm with Prometheus exporters enabled.  For the past few days I had this alert stuck as an error:

Query: `argocd_app_sync_total{phase!="Succeeded"}` which returns one application:
```
argocd_app_sync_total{container="application-controller", dest_server="https://kubernetes.default.svc", endpoint="http-metrics", exported_namespace="argocd", instance="10.42.2.194:8082", job="argocd-application-controller-metrics", name="grafana-dashboards", namespace="argocd", phase="Error", pod="argocd-application-controller-0", project="default", service="argocd-application-controller-metrics"}
```

However, this application is not in an error state.  Using the ArgoCD GUI, it shows:
```
OPERATION: Sync
PHASE: Succeeded
MESSAGE: successfully synced (all tasks run)
STARTED AT: 2 days ago (Fri Jul 08 2022 14:37:14 GMT-0400)
DURATION: 00:19 min
FINISHED AT: 2 days ago (Fri Jul 08 2022 14:37:33 GMT-0400)
```

Then I noticed that I have two entries for this application.  One that is "Error" (above) and one that is "Succeeded".

Query: `argocd_app_sync_total{name="grafana-dashboards"}`
```
argocd_app_sync_total{container="application-controller", dest_server="https://kubernetes.default.svc", endpoint="http-metrics", exported_namespace="argocd", instance="10.42.2.194:8082", job="argocd-application-controller-metrics", name="grafana-dashboards", namespace="argocd", phase="Succeeded", pod="argocd-application-controller-0", project="default", service="argocd-application-controller-metrics"}
```

Don't understand why ArgoCD  would create two of these in different Phase or not cleanup / reconcile / something... bug?

**Comment by roberto-civitas** [comment]
Having this issue as well

**Comment by aliceburigo** [comment]
We are facing the same problem.
We set up an alert using this metric comparing the percentage of sync errors and it was triggering false alerts.
We even had this metric recorded for an application where the name was updated weeks ago, with the label for both names over several days.
The metric doesn't seem to update after the last sync...
